activerecord/lib/active_record/associations/has_many_association.rb

```diff
@@ -84,7 +84,10 @@ def count_records
           # If there's nothing in the database and @target has no new records
           # we are certain the current target is an empty array. This is a
           # documented side-effect of the method that may avoid an extra SELECT.
```

Comment:
If my previous suggestion works, we'll need to update this comment ("target is an empty array") too.

Reply:
Good call, updated!
https://github.com/rails/rails/blob/db0e0c02cdbe6e69d88ce07e39eac689b0639417/activerecord/lib/active_record/associations/has_many_association.rb#L84-L86